GG/GG/Font.h

```diff
@@ -465,33 +465,19 @@ class GG_API Font
     void ProcessTagsBefore(const std::vector<LineData>& line_data, RenderState& render_state,
                            std::size_t begin_line, CPSize begin_char) const;
 
-    /** Returns the maximum dimensions of the string in x and y, and populates
-        \a line_data. */
-    Pt   DetermineLines(const std::string& text, Flags<TextFormat>& format, X box_width,
-                        std::vector<LineData>& line_data) const;
-
-    /** Returns the maximum dimensions of the string in x and y, and populates
-        \a line_data and \a text_elements.  Note that \a text_elements must be
-        empty. */
-    Pt   DetermineLines(const std::string& text, Flags<TextFormat>& format, X box_width,
-                        std::vector<LineData>& line_data,
-                        std::vector<boost::shared_ptr<TextElement> >& text_elements) const;
+    /** Populate \p text_elements with TextElements parsed from \p text. The only \p format flag
+        used is FORMAT_IGNORETAGS to determine if tags are ignored. */
+    std::vector<boost::shared_ptr<TextElement> > ComputeTextElements(
+        const std::string& text, const Flags<TextFormat>& format = FORMAT_NONE) const;
 
     /** Returns the maximum dimensions of the string in x and y, and populates
-        \a line_data.  The contents of \a text_elements will be used, and the
-        equivalent work done by DetermineLines() will be skipped.  Supplying a
+        \a line_data.  The contents of \a text_elements will be used.  Supplying a
         \a text and a \a text_elements that are incompatible will result in
         undefined behavior. */
     Pt   DetermineLines(const std::string& text, Flags<TextFormat>& format, X box_width,
```

Comment:
> If the function decides where linebreaks are, then maybe call it BreakTextIntoLines or DetermineLinebreakLocationsForText ?

The purpose isn't limited to line breaking, look into the implementation.

1. Break text at the target `box_width` depending on `format`.
2. Text justification depending on `format` (delegating the implementation details to `SetJustification`).

You can see it as you want, but you pour in TextElement s, a document width and justification rules and get a [typesetting], which is the very meaning of text layouting.

[typesetting]: https://en.wikipedia.org/wiki/Typesetting